src/cudamatrix/cu-matrix-test.cc

```diff
@@ -1430,6 +1430,10 @@ static void UnitTestCuMatrixAddVecVec() {
 
 template<typename Real>
 static void UnitTestCuMatrixAddMatMatBatched() {
+  // Random stride is disabled as AddMatMatBatched requires consistent stride
+#if HAVE_CUDA == 1
+  CuDevice::Instantiate().SetRandomStrideMode(false);
```

Comment:
`bool old_mode = CuDevice::Instantiate().SetRandomStrideMode(false);`

Then at line 1500, do:

`CuDevice::Instantiate().SetRandomStrideMode(old_mode);`

This code right now depends on the assumption that the old mode was set to true. This is not necessarily true, as it depends on when this test function is called.